lib/snapshot.js

```diff
@@ -379,31 +379,42 @@ function saveAsUnsupportedContent( filename, content_type ) {
 
 }
 
+/**
+ * Run a job from the queue if there is any, and the worker is not busy.
+ */
 function check_site_queue() {
-	if ( false === downloading ) {
-		if ( reload_config ) {
-			reload_config = false;
-			_blacklist.loadConfig();
-			logger.debug( process.pid + ': blacklist file reloaded' );
-		}
-		if ( site_queue.length > 0 ) {
-			downloading = true;
-			logger.debug( process.pid + ': site queue size = ' + site_queue.length );
-
-			var s_details = site_queue[0];
-			site_queue.shift();
-			logger.debug( process.pid + ': snapping: ' + s_details.url );
-			take_snapshot(
-				s_details.url,
-				s_details.file,
-				s_details.width,
-				s_details.height,
-				s_details.screenWidth,
-				s_details.screenHeight,
-				s_details.scaleFactor
-			);
-		}
+	if ( downloading || site_queue.length === 0 ) {
```

Comment:
probably don't need the `downloading` check here since it's also done in `check_site_queue()` https://github.com/Automattic/mShots/pull/56/files#diff-5ec71a0778abedc1f8e126438fd6bca0551fe359a8c68705c73eb27ea5497155L433